streams/jvm/src/test/scala/zio/stream/StreamSpec.scala

```diff
@@ -1456,4 +1460,31 @@ class ZStreamSpec(implicit ee: org.specs2.concurrent.ExecutionEnv)
       .either
       .map(_ must_=== Left("Ouch"))
   }
+
+  private def zipWithLatest = unsafeRun {
+    // In case this test fails, try increasing delay time propotionally.
+    val s1     = Stream("s1", "s2", "s3").tap(_ => ZIO.sleep(1200.milliseconds))
```

Comment:
Let's rewrite this test to use latches instead of sleeps, as sleeps will definitely cause flakiness in the CI. Let me know if you need help with that.